learn/guides/managing-dependencies.md

```diff
@@ -12,80 +12,163 @@ redirect_from:
+The module name of the default module is always the package name. 
 
-**Note:** The source code of the `main.bal` file in your package will have the import statement below.
+The following example shows how we can import modules from the `ballerina/io` package.
 
 ```bal
-import ballerina/io;
+import ballerina/io;`// imports default module of io package
+
+public function main() {
+	io:println("Hello world!");
+}
 ```
+The following is another example that shows the usage of multiple modules from different libraries.
 
-You can import a module by providing the organization name and the module name. The module name of the default module is always the package name.
+```bal
+// imports the default module from the ballerina/log package
+import ballerina/log;
+// imports the default module from googleapis.gmail package with an import prefix
+import ballerinax/googleapis.gmail as gmail;
+// imports the only non-default module from googleapis.gmail package with an import prefix
+import ballerinax/googleapis.gmail.'listener as gmailListener;
+
+configurable string refreshToken = ?;
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable int port = ?;
+configurable string project = ?;
+configurable string pushEndpoint = ?;
+
+gmail:ConnectionConfig gmailConfig = {
+    auth: {
+        refreshUrl: gmail:REFRESH_URL,
+        refreshToken: refreshToken,
+        clientId: clientId,
+        clientSecret: clientSecret
+        }
+};
+
+listener gmailListener:Listener gmailEventListener = new(port, gmailConfig, project, pushEndpoint);
+
+service / on gmailEventListener {
+   remote function onNewEmail(gmail:Message message) returns error? {
+           log:printInfo("New Email : " , message = message);
+   }   
+}
 
-### Importing Modules of the Same Package
+```
 
-The organization name can be omitted only if you are importing a module from the same package. 
+## Specifying dependency versions
 
-This is the directory structure of your package after adding the `hello_world.util` module.
+The dependency versions are automatically figured out by the compiler when a package is built. The latest compatible versions of the dependencies are resolved by searching the packages of dependencies in the distribution repository and the Ballerina Central repository and if specified from the local repository as well.
 
-```bash
-.
-├── Ballerina.toml
-├── main.bal
-└── modules
-     └── util
-         └── util.bal
-
-2 directories, 3 files
-```
+When you execute `bal build` for the first time on the package, The CLI operation will auto-generate the `Dependencies.toml` in the package root. 
+This will contain the resolved dependency versions. From thereon, the versions locked in the `Dependencies.toml` are considered as the minimum required versions for subsequent builds.
```

Comment:
```suggestion
This will contain the resolved dependency versions. From thereon, the versions locked in the `Dependencies.toml` are considered as the minimum required versions for the subsequent builds.
```